Create dashboard ui and chart framework

isabelt220 opened the pull request:
I added the chart framework to the front-end and got it to work with some dummy data. I'll be working on connecting it with the back-end to get the data from the database and render it in the chart next.
The chart looks ugly right now and the sizing are off but I'll work on those a bit later.
<img width="1280" alt="Screen Shot 2019-11-27 at 5 36 00 PM" src="https://user-images.githubusercontent.com/43298030/69763931-bc1dc900-113c-11ea-8f85-5666971ef3be.png">

isabelt220 commented:
I got the chart to fetch data from the back-end! Here's an update on how it looks right now. I'll be working on getting the chart to get data from the back-end based on which month users want to look at later. Also, I'll be fixing the x-axis and resizing everything as well.
<img width="1280" alt="Screen Shot 2019-11-28 at 11 09 54 AM" src="https://user-images.githubusercontent.com/43298030/69820665-a1e0fb00-11cf-11ea-91ad-d9c177c44cf8.png">

bonnieli commented on client/src/pages/TotalExpensesChart.js:
```diff
@@ -0,0 +1,52 @@
+
+class TotalExpensesChart extends Component {
+  state = {
+      daily_expenses: [{expense: 10.01, date: "Nov 3"},
```
do you need this mock data now that you're fetching?

isabelt220 replied:
I'll remove it later!

bonnieli commented on client/src/pages/TotalExpensesChart.js:
```diff
@@ -0,0 +1,52 @@
+} from "@devexpress/dx-react-chart-material-ui";
+
+const useStyles = theme => ({
+  root: {
```
doesn't look like you're using this style here

isabelt220 replied:
I'm working on the styling now! :)

bonnieli commented on client/src/pages/TotalExpensesChart.js:
```diff
@@ -0,0 +1,52 @@
+  }
+
+  componentDidMount() {
+      fetch("/receipts/daily-expenses/2019/11")
```
maybe by default you can find out the current year and current month?

isabelt220 replied:
you're right! I'll have a default value for the chart data added in!

bonnieli commented on client/src/pages/TotalExpensesChart.js:
```diff
@@ -0,0 +1,52 @@
+      })
+      .then(results => {
+          this.setState({
+              daily_expenses: results.data
```
best to use camelCase for react :) `dailyExpenses` as your state

isabelt220 replied:
okay got it!

isabelt220 commented:
I finished all the functional stuff for the total expense chart and the top categories, so everything is working. I should only have the UI styling left to do, but let me know if I missed anything!
<img width="1280" alt="Screen Shot 2019-11-29 at 1 21 42 PM" src="https://user-images.githubusercontent.com/43298030/69885455-43894a80-12ab-11ea-8ef9-ac32a47c17b1.png">

isabelt220 commented:
I updated and fixed most of the styling for the dashboard UI and everything should be working fine. The only thing I haven't fixed is the x-axis label for the chart. The labels are still overlapping and I haven't found a way to fix it yet, so any help on how to fix the label overlapping would be appreciated!
<img width="1280" alt="Screen Shot 2019-12-02 at 11 32 19 AM" src="https://user-images.githubusercontent.com/43298030/69976705-6f007500-14f7-11ea-8200-000d5d30f6d5.png">

bonnieli commented on server/api/category_controller.py:
```diff
@@ -0,0 +1,64 @@
+    receipts = (Receipt.query
+                .filter(Receipt.receipt_date <= end_date)
+                .filter(Receipt.receipt_date >= start_date)
+                .order_by(Receipt.category.desc()))
```
you should take a look at the SQL aggregate func :D https://www.postgresql.org/docs/9.5/tutorial-agg.html

this will probably be a lot more efficient and much less lines of code

bonnieli replied:
it'll probably be along the lines of 

```
SELECT category, SUM(amount)
FROM Receipt
GROUP BY category
ORDER BY SUM(amount)
```

written in the python query way 

if possible, you probably won't even need to manually calculate it :)

isabelt220 replied:
great! I'll go take a look at it later!

bonnieli commented:
some merge conflicts, need to update branch

isabelt220 commented:
I reverted the merge just now cause it was giving me problems while I was trying to fix the x-axis label, but I'll resolve the in a bit right after I'm done with this. So I fixed the x-axis label to show only the dates for the month. Let me know if this looks okay!
<img width="1280" alt="Screen Shot 2019-12-02 at 1 39 59 PM" src="https://user-images.githubusercontent.com/43298030/69985664-ae37c180-1509-11ea-83c2-f54999ab9c33.png">
